include/igl/randperm.h

```diff
@@ -9,18 +9,23 @@
 #define IGL_RANDPERM_H
 #include "igl_inline.h"
 #include <Eigen/Core>
+#include <random>
 namespace igl
 {
   // Like matlab's randperm(n) but minus 1
   //
   // Inputs:
   //   n  number of elements
+  //   urbg An instance of UnformRandomBitGenerator. When not given,
+  //        randperm will use default random bit generator std::minstd_rand
+  //        initialized with random seed generated by std::rand()
   // Outputs:
   //   I  n list of rand permutation of 0:n-1
-  template <typename DerivedI>
+  template <typename DerivedI, typename URBG=std::minstd_rand>
   IGL_INLINE void randperm(
     const int n,
-    Eigen::PlainObjectBase<DerivedI> & I);
+    Eigen::PlainObjectBase<DerivedI> & I,
+    URBG urbg=std::minstd_rand(std::rand()));
```

Comment:
Does this default initialization work when `URBG` is a different type ? Otherwise, could we write `URBG urbg = URBG(std::srand())` ? Another option is to use a function overload to define the optional argument, which is maybe more in line with libigl's style.

Reply:
Thank you for the suggestion. I think using function overload is a better choice.